src/pages/NoteList.tsx

```diff
@@ -1,17 +1,74 @@
+/* eslint-disable indent */
+import { SetStateAction, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 import styled from 'styled-components';
 
 import Button from '../components/Button';
-import NoteItemList from '../components/NoteListItem';
+import DropDown from '../components/DropDown';
+import NoteListItem, { NoteItemListProps } from '../components/NoteListItem';
+import { memoDataType } from '../types';
 
 const NoteList = () => {
+  const navigate = useNavigate();
+  const clickHandler = (link: string) => {
+    navigate(`/${link}`);
+  };
+
+  const [dropDownText, setDropDownText] = useState('최근 생성순');
+
+  const handleDropDownClick = (event: { target: { value: SetStateAction<string> } }) => {
+    setDropDownText(event.target.value);
+  };
+
+  const initialData = JSON.parse(localStorage.getItem('dataList')!);
+  const [memoList, setMemoList] = useState(initialData);
+
+  useEffect(() => {
+    switch (dropDownText) {
+      case '북마크':
+        setMemoList(memoList.filter((memo: memoDataType) => memo.bookMark));
+        break;
+      case '최근 생성순':
+        setMemoList(initialData.reverse());
+        break;
+      case '최근 수정순':
+        setMemoList(initialData.sort());
+        console.log(memoList);
+        break;
+    }
+  }, [dropDownText]);
+
+  const handleBookMark = (index: number, event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.stopPropagation();
+    const tempList = [...memoList];
+    tempList[index].bookMark = !tempList[index].bookMark;
+    setMemoList(tempList);
+    localStorage.setItem('dataList', JSON.stringify(memoList));
+    console.log(JSON.parse(localStorage.getItem('dataList')!));
+  };
+
+  const noteList = memoList.map((data: NoteItemListProps, index: number) => (
+    <NoteListItem
+      key={index}
+      index={index}
+      title={data.title}
+      bookMark={data.bookMark}
+      handleBookMark={handleBookMark}
+      fixTime={data.fixTime}
+    />
```

Comment:
```suggestion
<NoteListItem
      key={index}
      index={index}
      handleBookMark={handleBookMark}
      {...data}
    />
```

웹으로만 수정한 코드라 이게 맞는지는 잘 모르겠지만... 위에서 설명한 것처럼 spread 문법을 이용하면 좋을 것 같아요!

그리고 저도 정우 오빠한테 배웠는데 index는 변경의 위험성이 있어서 key값으로 주는 건 좋지 않다고 해요!! 현재 고유 id가 없어서 index로 주신 것 같은데 다음에는 id도 같이 생성해 주면 더 좋을 것 같아요!